Library/Formula/gcc.rb

```diff
@@ -38,13 +38,25 @@ def osmajor
   # enabling multilib on a host that can't run 64-bit results in build failures
   option "without-multilib", "Build without multilib support" if MacOS.prefer_64_bit?
 
-  depends_on "gmp"
-  depends_on "libmpc"
-  depends_on "mpfr"
-  depends_on "cloog"
-  depends_on "isl"
   depends_on "ecj" if build.with?("java") || build.with?("all-languages")
 
+  resource "gmp" do
+    url "#{Formula["gmp"].stable.url}"
+    sha1 "#{Formula['gmp'].stable.checksum}"
+  end
+  resource "mpfr" do
+    url "#{Formula["mpfr"].stable.url}"
+    sha1 "#{Formula["mpfr"].stable.checksum}"
+  end
+  resource "libmpc" do
+    url "#{Formula["libmpc"].stable.url}"
+    sha1 "#{Formula["libmpc"].stable.checksum}"
+  end
+  resource "isl" do
+    url "#{Formula["isl"].stable.url}"
+    sha1 "#{Formula["isl"].stable.checksum}"
+  end
```

Comment:
Expand the URLs and checksums rather than reach into formula objects from the DSL.